amida-tech/indaba-client
# Feature/indaba 959 search grommet

**Must be run with INBA-969, under greyscale.**

#### What does this PR do?
When in Create Project Wizard or the Project Management, you now can choose and duplicate an existing group from any other project. This copies over the users and the name, but is stopped if a name already exists in your project, encouraging renaming. 

#### Related JIRA tickets:
INBA-959
INBA-969

#### How should this be manually tested?
Assuming you already have a few projects with groups and users assigned to them...

1. Go into Project Management, then go to the Users tab.
2. Go to the User Groups tab.
3. Click on the search bar with placeholder "Copy an existing user group..." You should get a drop down of all the groups in your system, complete with hints as to which project and how many users are in it. (Screenshot provided.)
4. Click away from the drop down. It should disappear.
5. Click on the search bar again. Type in the start of a filter. The names should parse down. 
6. Click on a name that is not in your project. Verify that the group AND the users are added. (Note, this should add users to your project if they weren't in there before.)
7. Click on the search bar again. Click on a name of a group that the same as another group in your project. The group should not appear, and you should get a toast pop up explaining why not.

Repeat the same steps in the Create Project Wizard.

#### Background/Context
I think it's time to start not relying on redux for UI handling unless there's a good reason, like you need to multiple sources to be able to flick a modal on and off. The excessive redux dispatches clutter the system and slow it down. For the time being I've left them in there, but I have eliminated the anonymous functions in the renders where I found them. 

#### Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/4026454/50948677-53b8f100-1471-11e9-9e05-a929bbd7fc08.png)
